airbyte-cdk/python/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py

```diff
@@ -400,7 +402,13 @@ def read_records(
         # * Why is this abstraction not on the DeclarativeStream level? DeclarativeStream does not have a notion of stream slicers and we
         #    would like to avoid exposing the stream state outside of the cursor. This case is needed as of 2023-06-14 because of
         #    interpolation.
-        slice_state = self.stream_slicer.select_state(stream_slice) if hasattr(self.stream_slicer, "select_state") else stream_state
+        if self.cursor and hasattr(self.cursor, "select_state"):
```

Comment:
do we still need the hasattr if we introduce the Cursor interface?

Reply:
`select_state` still doesn't make sense as part of DatetimeBasedCursor and therefore I'm not so sure how it'll evolve as part of the cursor interface. If you think otherwise, I don't mind changing this though